vllm/engine/llm_engine.py

```diff
@@ -711,3 +717,42 @@ def _run_workers(
         for other_output in all_outputs[1:]:
             assert output == other_output
         return output
+
+    def _execute_model_dag(
+        self,
+        seq_group_metadata_list=None,
+        blocks_to_swap_in=None,
+        blocks_to_swap_out=None,
+        blocks_to_copy=None,
+        get_all_outputs: bool = False,
+    ) -> Any:
+        """Runs the given method on all workers using static DAG APIs."""
+        from ray.dag import OutputNode, InputNode
+        assert self.parallel_config.worker_use_ray
+
+        all_outputs = []
+        with InputNode() as input_data:
+            forward_dag = OutputNode([
```

Comment:
Should cache this?

Reply:
Yeah, and we should do some testing that it works properly once cached.

Reply:
Ah yeah good point. I will update it

Reply:
done. test_models.py execute models many times, and it passes, meaning it verifies it works.